src/plugins/intel_gna/src/gna2_model_export_helper.cpp

```diff
@@ -278,3 +215,40 @@ void ExportTlvModel(uint32_t modelId,
         THROW_GNA_EXCEPTION << "Not succesfull status returned: " << tlv_status << ", from Gna2ExportTlv() function\n";
     }
 }
+
+void* ExportSueLegacyUsingGnaApi2(uint32_t modelId, uint32_t deviceIndex, Gna2ModelSueCreekHeader* modelHeader) {
+    uint32_t exportConfig;
+    auto status = Gna2ModelExportConfigCreate(gnaUserAllocatorAlignedPage, &exportConfig);
+    GNADeviceHelper::checkGna2Status(status, "Gna2ModelExportConfigCreate");
+
+    status = Gna2ModelExportConfigSetSource(exportConfig, deviceIndex, modelId);
+    GNADeviceHelper::checkGna2Status(status, "Gna2ModelExportConfigSetSource");
+    status =
+        Gna2ModelExportConfigSetTarget(exportConfig, static_cast<Gna2DeviceVersion>(DeviceVersion::GNAEmbedded1_0));
+    GNADeviceHelper::checkGna2Status(status, "Gna2ModelExportConfigSetTarget");
+
+    void* bufferSueCreekHeader = nullptr;
+    uint32_t bufferSueCreekHeaderSize;
+
+    status = Gna2ModelExport(exportConfig,
+                             Gna2ModelExportComponentLegacySueCreekHeader,
+                             &bufferSueCreekHeader,
+                             &bufferSueCreekHeaderSize);
+    GNADeviceHelper::checkGna2Status(status, "Gna2ModelExport(LegacySueCreekHeader)");
+
+    (*modelHeader) = *(reinterpret_cast<Gna2ModelSueCreekHeader*>(bufferSueCreekHeader));
```

Comment:
shouldn't we check the pointer? It may lead to the segfault.

Reply:
This is out of scope of this PR.